File: src/commands/verify.js
const config = require('../config/config.json').commands.verify;

function roles(message) {
  const member = message.mentions.members.first();
  const guild = message.guild;
  const roleToAdd = guild.roles.find((role) => role.name === config.roleToAdd);
  const roleToRemove = guild.roles.find((role) => role.name === config.roleToRemove);

  member.addRole(roleToAdd);
  member.removeRole(roleToRemove);
}

function welcomeMessages(message) {
  const member = message.mentions.members.first();
  const guild = message.guild;

  const botsChannel = guild.channels.get(config.botsChannelId);
  const botsMessage = config.botsChannelMessage.replace('<user>', `${member.user}`);
  botsChannel.send(botsMessage);

  const generalChannel = guild.channels.get(config.generalChannelId);
  const generalMessage = config.generalChannelMessage.replace('<user>', `${member.user}`);
  generalChannel.send(generalMessage);
}

function deleteMessages(channel, deleteAmount) {
  channel.bulkDelete(deleteAmount).then((messages) => {
    channel.send(`${messages.size - 1} messages deleted.`).then((mess) => {
      setTimeout(() => {
        mess.delete();
      }, 5000);
    });
  });
}

module.exports = {
  name: 'verify',
  description: 'Verifies the mentioned user',
  usage: 'verify <@user> [num_message_to_delete]',
  permission: config.permission,
  execute: (message, args) => {
    if (!message.member.roles.find((role) => role.name === config.permission)) return;

    roles(message);
    welcomeMessages(message);

    if(args[1] && parseInt(args[1])) {
      setTimeout(() => {
        deleteMessages(message.channel, parseInt(args[1]) + 1);
      }, 5000);
    }
  },
};

File: lib/schwarznasenschaf/commands/verify.rb
# frozen_string_literal: true

module Schwarznasenschaf
  module Commands
    module Verify
      extend Discordrb::Commands::CommandContainer

      command_attributes = {
        description: 'Verifies the mentioned user and optionally clears chat',
        help_available: true,
        max_args: 2,
        min_args: 1,
        rescue: 'An  error occured while trying to execute this command.',
        required_roles: [Support::Config::ROLES[:mod_team]],
        usage: 'verify [mentioned_user] <messages_to_clear>'
      }

      command :verify, command_attributes do |event, *args|
        user = event.message.mentions.first
        return unless user

        set_roles user, event.server
        send_bot_channel_messages user, event.server
        send_general_channel_messages user, event.server

        return if args.last.to_i.zero?

        result = Commands.bulk_delete event.channel, args.last.to_i
        Commands.send_and_delete event.channel, result
      end

      def self.set_roles(user, server)
        member = server.member user.id

        role_to_add = server.roles.detect do |role|
          role.id == Support::Config::ROLES[:verified]
        end
        role_to_remove = server.roles.detect do |role|
          role.id == Support::Config::ROLES[:just_joined]
        end

        member.add_role role_to_add
        member.remove_role role_to_remove
      end

      def self.send_bot_channel_messages(user, server)
        bots_channel = server.channels.detect do |channel|
          channel.id == Support::Config::CHANNELS[:bots]
        end

        bots_message = "#{user.mention} you can assign yourself roles here: "\
                       "https://roleypoly.com/s/#{server.id}"

        bots_channel.send bots_message
      end

      def self.send_general_channel_messages(user, server)
        general_channel = server.channels.detect do |channel|
          channel.id == Support::Config::CHANNELS[:general]
        end

        general_message = "Welcome #{user.mention}!"

        general_channel.send general_message
      end
    end
  end
end
